### PR TITLE
[ECO-1540] Update architecture specification

### PR DESCRIPTION
**Pull request by alnoki**
# Description

Add assorted architecture specifications per recent engineering discussions

# Testing

Passes `pre-commit`

# Checklist

- [x] Did you update relevant documentation?
- [x] ~Did you add tests to cover new code or a fixed issue?~
- [x] ~Did you update the changelog?~
- [x] Did you check off all checkboxes from the linked Linear task?

### CODE REVIEW

File: doc/architecture-specification.md [hunk 1]
```diff
@@ -30,15 +30,47 @@ These keywords `SHALL` be in `monospace` for ease of identification.
...
+   lapsed.
+1. A view function shall enable enable lookup of global total value locked,
+   number of emojicoin markets, and cumulative volume, via aggregators when
+   necessary.
+1. Events to simulate swap and liquidity provisioning operations `SHALL` be
+   calculated via simulation functions that can be called directly from view
+   functions.
+1. Simulation functions `SHALL` be paired with simulation commitment functions
+   such that results can be applied during runtime, using the same logic as
+   view function indexing functions.
```
Comment (xbtmatt) [inline doc/architecture-specification.md:51]: Is this essentially specifying that the underlying logic for the order flow and any effects it has on liquidity will be abstracted out to a view function? Just curious what this means here, trying to make sure I understand

**Review by xbtmatt — APPROVED** [review]
Looks good to me, and I love the `emos` subunits :)

Feel free to resolve the question I had if you want to merge it ASAP, I'm mostly just curious

File: doc/architecture-specification.md (hunk 1 above)
Comment (alnoki) [reply doc/architecture-specification.md:51]: Yes, the idea is that the view function will rely on the same logic as the executor function.

It's similar to blockSTM in a prepare/commit scheme

Comment (alnoki) [reply doc/architecture-specification.md:51]: See here: https://github.com/econia-labs/emojicoin-dot-fun/blob/eb0e84fef21605f64429562eb2b1e99b91f77234/src/move/sources/emojicoin_dot_fun.move#L111

**Pull request merged**